File: server/router/cloudpods/image/imageRepos.go
package image

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type ImageReposRouter struct {
}

func (s *ImageReposRouter) InitImageReposRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/image/imageRepos")
	imageApi := v1.ApiGroupApp.CloudApiGroup.ImageApiGroup.ImageReposApi
	{

		dataRouter.POST("getImageReposList", imageApi.GetImageReposList)           // 获取镜像仓库列表
		dataRouter.POST("addImageRepos", imageApi.AddImageRepos)                   // 新增镜像仓库
		dataRouter.POST("deleteImageRepos", imageApi.DeleteImageRepos)             // 删除镜像仓库
		dataRouter.POST("deleteImageReposByIds", imageApi.DeleteImageReposByIds)   // 删除镜像仓库
		dataRouter.POST("getImageReposImageList", imageApi.GetImageReposImageList) // 获取镜像仓库的镜像列表
	}
}

File: web/src/api/cloudpods/image/imageRepos.js
import request from '@/utils/request'

// 获取镜像仓库列表
export function getImageReposList(params) {
    return request({
        url: 'cloud/image/imageRepos/getImageReposList',
        method: 'post',
        data: params
    })
}
// 新增镜像仓库
export function addImageRepos(params) {
    return request({
        url: 'cloud/image/imageRepos/addImageRepos',
        method: 'post',
        data: params
    })
}
// 删除镜像仓库
export function deleteImageRepos(params) {
    return request({
        url: 'cloud/image/imageRepos/deleteImageRepos',
        method: 'post',
        data: params
    })
}
// 批量删除镜像仓库
export function deleteImageReposByIds(params) {
    return request({
        url: 'cloud/image/imageRepos/deleteImageReposByIds',
        method: 'post',
        data: params
    })
}
// 获取镜像仓库的镜像列表
export function getImageReposImageList(params) {
    return request({
        url: 'cloud/image/imageRepos/getImageReposImageList',
        method: 'post',
        data: params
    })
}